specs/ClientCertificate.md

```diff
@@ -0,0 +1,380 @@
+# Background
+The WebView2 team has been asked for an API to intercept client certificates when WebView2 is
+making a request to an Http server that needs a `client certificate` for `Http authentication`.
+This event allows you to show ui if desired, replace default client certificate dialog prompt,
+programmatically query the certificates and select a certificate from the list to respond to the server.
+
+In this document we describe the API. We'd appreciate your feedback.
+
+
+# Description
+We propose adding `ClientCertificateRequested` API that allows you to query the mutually trusted CA
+certificates, and select the certificate from the list to respond to the server.
+
+When the event is raised, WebView2 will pass a `CoreWebView2ClientCertificateRequestedEventArgs`, which lets
+you to view the server information that requested client certificate authentication, list of certificates filtered
+from issuers that the server trusts, inspect the certificate metadata and serveral options for responding to client certificate request.
+
+- You can select a certificate from the list to respond to the server.
+- You can choose to respond to the server without a certificate.
+- You can choose to display default client certificate selection dialog prompt to let user to respond to the server.
+- You can cancel the request.
+
+# Examples
+The following code snippets demonstrate how the ClientCertificateRequested event can be used:
+## Win32 C++
+```cpp
```

Comment:
Do we have enough info in our objects to allow the end dev to also interact with the OS cert APIs? Is there a valid scenario for interacting with OS cert APIs, and if so, should that be another sample or worked into an existing sample?

Reply:
I don't think there is valid scenario to interact with OS API's. We are only going to expose list of filtered/mutually trusted certificates to the end dev (same as certificates we show in our dialog prompt currently). We are not performing any other functionalities apart from what browser does.